bin/cylc-cat-log

```diff
@@ -29,227 +29,271 @@ if remrun().execute():
+        basename))
+
+
+def get_task_job_log_path(
+        options, suite, point, task, submit_num, user_at_host):
+    """Return file name of a task job log, given the options."""
+    if user_at_host and "@" in user_at_host:
+        owner, host = user_at_host.split("@", 1)
+    elif user_at_host:
+        owner, host = (None, user_at_host)
+    else:
+        owner, host = (None, None)
+    if options.list_mode:
+        basename = "."
+    elif options.filename:
+        try:
+            basename = JOB_LOG_DEST_MAP[options.filename]
+        except KeyError:
+            basename = options.filename
+    else:
+        basename = JOB_LOG_DEST_MAP[NAME_DEFAULT]
+    if submit_num != "NN":
+        submit_num = "%02d" % submit_num
+    return os.path.normpath(os.path.join(
+        GLOBAL_CFG.get_derived_host_item(
+            suite, "suite job log directory", host, owner),
+        point, task, submit_num, basename))
+
+
+def get_task_job_attrs(options, suite, point, task, submit_num):
+    """Return (user@host, command0) of a task job log.
+
+    user@host is set if task job is run remotely and for relevant log files.
+    command0 is set if task job if running on a batch system that requires a
+    special command to view stdout/stderr files.
+
+    """
+    if (options.filename in JOB_LOG_LOCAL_ALWAYS or
+             options.list_mode == LIST_MODE_LOCAL):
+        return (None, None)
+    suite_dao = CylcSuiteDAO(
+        os.path.join(
+            GLOBAL_CFG.get_derived_host_item(suite, "suite run directory"),
+            CylcSuiteDAO.DB_FILE_BASE_NAME),
+        is_public=True)
+    task_job_data = suite_dao.select_task_job(None, point, task, submit_num)
+    suite_dao.close()
+    if task_job_data is None:
+        return (None, None)
+    if "@" in task_job_data["user_at_host"]:
+        owner, host = str(task_job_data["user_at_host"]).split("@", 1)
+    else:
+        owner, host = (None, str(task_job_data["user_at_host"]))
     user_at_host = None
-    suite_run_dir = GLOBAL_CFG.get_derived_host_item(
-        suite, "suite run directory")
-
-    if len(args) == 1:
-        # Suite logs.
-        if any([options.job_activity, options.job_diff, options.job_status,
-                options.list_local or options.list_remote]) or (
-                options.job_custom is not None or
-                options.subnum != "NN"):
-            parser.error("Task log option(s) are not legal for suite logs.")
-
-        log_dir = os.path.join(suite_run_dir, "log", "suite")
-        if options.stdout:
-            fpath = os.path.join(log_dir, "out")
-        elif options.stderr:
-            fpath = os.path.join(log_dir, "err")
+    if is_remote_host(host) or is_remote_user(owner):
+        if host and owner:
+            user_at_host = owner + "@" + host
+        elif host:
+            user_at_host = host
+        elif owner:
+            user_at_host = owner + "@localhost"
+    if (options.list_mode or
```

Comment:
I want to continue only if I am not in list or location mode, and the file is "job.out" or "job.err" while the job is running, and the batch system and the batch system job ID are known. Yes, it is complicated.